File: recurplot.m
function y=recurplot(x,n)

y=[];
recur(x);

%Remove adjacent duplicates, there are some issues in running recursive functions in a
%forloop.
y(diff(y)==0)=[];

    function recur(x)
        if x==1
            y(end+1)=1;
            return
        end
        
        for i=1:n
            y(end+1)=x;
            recur(x-1)
            y(end+1)=x;
        end
        %% Can also set depth of recursion this way.
        %         y(end+1)=x;
        %         recur(x-1)
        %         y(end+1)=x;
        %         recur(x-1)
        %         y(end+1)=x;
    end
%% Plotting
plot(y, '-o')
ylabel('X value')
xlabel('Stack Step')
title(strcat('Stack:', num2str(x), '   Recursive Functions:', num2str(n)))
end

File: recurplot.py
import matplotlib.pyplot as plt

def recurplot(x,n):
	y=[]
	print(x,n)

	def recur(x):
		if x==0:
			#print(x)
			y.append(0)
			return

		for i in range(n):
			y.append(x)
			#print(x)
			recur(x-1)
			#print(x)
			y.append(x)
	
	recur(x)
	
	#still have to remove adjacent duplicates
	y_nodupe = [x for i, x in enumerate(y) if i==0 or x!=y[i-1]]

	#Plotting
	plt.plot(y_nodupe, '-o')
	plt.xlabel('Stack Step')
	plt.ylabel('X Value')
	plt.show()

	return y, y_nodupe